rts/lra/lra-filters/src/main/java/io/narayana/lra/filter/ClientLRAResponseFilter.java

```diff
@@ -36,16 +36,19 @@
     public void filter(ClientRequestContext requestContext, ClientResponseContext responseContext) throws IOException {
         Object incomingLRA = responseContext.getHeaders().getFirst(LRA_HTTP_HEADER);
 
-        if (incomingLRA == null) {
-            incomingLRA = requestContext.getProperty(LRA_HTTP_HEADER);
-        }
-
         /*
          * if the incoming response contains a context make it the current one
-         * (note we never popped the context in the request filter so we don't need to push outgoingLRA
+         * otherwise check if the outgoing request contained a context then
+         * put it back on the thread.
          */
         if (incomingLRA != null) {
             Current.push(new URL(incomingLRA.toString()));
+        } else {
+            Object outgoingLRA = requestContext.getHeaders().getFirst(LRA_HTTP_HEADER);
+
+            if (outgoingLRA != null) {
+                Current.push((URL) outgoingLRA); // put it back on the request
+            }
```

Comment:
The origin comment says
```
note we never popped the context in the request filter so we don't need to push outgoingLRA
```
So it does pop the context in the request filter now ? I can not find it in the ClientLRARequestFilter.